airflow/kubernetes/pod_launcher_deprecated.py

```diff
@@ -306,15 +306,15 @@ def process_status(self, job_id, status):
         """Process status information for the job."""
         status = status.lower()
         if status == PodStatus.PENDING:
-            return State.QUEUED
+            return TaskInstanceState.QUEUED
         elif status == PodStatus.FAILED:
             self.log.error("Event with job id %s Failed", job_id)
-            return State.FAILED
+            return TaskInstanceState.FAILED
         elif status == PodStatus.SUCCEEDED:
             self.log.info("Event with job id %s Succeeded", job_id)
-            return State.SUCCESS
+            return TaskInstanceState.SUCCESS
```

Comment:
I'm not so sure about this section, I'll have to dig a little deeper. But using the pod state being succeeded to assume TIState was success isn't right, TIState could have also failed!

Reply:
Same concerns in a lot of this file, to be honest.

Reply:
This could be another instance where we should better introduce a new type, I’ll revert this for now as well.